src/Alignment.cpp

```diff
@@ -372,134 +372,138 @@ int Alignment::numExposedEdges(const Graph& G1, const Graph& G2) const {
+          }
         }
-        //std::cout << "789-----------------789-----------789-------------789" << std::endl;
-    }else{
-	ret += G2Matrix[A[node1]][A[node2]] + 1;
-        cerr << "numerator_type not specified, Using default numerator." << endl;
-    }
+        return ret;
     
-    return ret;
 }
 
 unsigned Alignment::multiS3Denominator(const Graph& G1, const Graph& G2) const {
-    uint ret = 0; 
-    vector<vector<uint> > G1EdgeList;
-    vector<vector<uint> > G2EdgeList;
-    G1.getEdgeList(G1EdgeList);
-    G2.getEdgeList(G2EdgeList);
-    Matrix<MATRIX_UNIT> G1Matrix;
-    Matrix<MATRIX_UNIT> G2Matrix;
-    G1.getMatrix(G1Matrix);
-    G2.getMatrix(G2Matrix);
-    uint node1, node2;
-    const uint n1 = G1.getNumNodes();
-    const uint n2 = G2.getNumNodes();
-    vector<uint> whichPeg(n2, n1); // value of n1 represents not used
+    uint ret = 0;
+        vector<vector<uint> > G1EdgeList;
+        vector<vector<uint> > G2EdgeList;
+        G1.getEdgeList(G1EdgeList);
+        G2.getEdgeList(G2EdgeList);
+        Matrix<MATRIX_UNIT> G1Matrix;
+        Matrix<MATRIX_UNIT> G2Matrix;
+        G1.getMatrix(G1Matrix);
+        G2.getMatrix(G2Matrix);
+        uint node1, node2;
+        const uint n1 = G1.getNumNodes();
+        const uint n2 = G2.getNumNodes();
+        vector<uint> whichPeg(n2, n1); // value of n1 represents not used
 
-    for (uint i = 0; i < n1; ++i){
-        whichPeg[A[i]] = i; // inverse of the alignment
-    }
-    
-    if (MultiS3::denominator_type == MultiS3::ee_global or MultiS3::denominator_type == MultiS3::rt_global){
-        for (uint i = 0; i < n2; ++i) for (uint j = 0; j < i; ++j){
-            bool exposed = (G2Matrix[i][j] > 0);
-            if(!exposed && whichPeg[i] < n1 && whichPeg[j] < n1 && G1Matrix[whichPeg[i]][whichPeg[j]] > 0){
-                exposed = true;
-            }
-            if(exposed and MultiS3::denominator_type == MultiS3::ee_global) ret++;
-            if(exposed and MultiS3::denominator_type == MultiS3::rt_global){
-                if (whichPeg[i] < n1 && whichPeg[j] < n1){
-                    ret += G1Matrix[whichPeg[i]][whichPeg[j]] + G2Matrix[i][j];
-                }else{
-                    ret +=  G2Matrix[i][j];
-                }
-            }        
+        for (uint i = 0; i < n1; ++i){
+            whichPeg[A[i]] = i; // inverse of the alignment
         }
-    }else if (MultiS3::denominator_type == MultiS3::ee_i or MultiS3::denominator_type == MultiS3::rt_i){
-        for (uint i = 0; i < n2; ++i) for (uint j = 0; j < i; ++j){
-            bool exposed = (whichPeg[i] < n1 && whichPeg[j] < n1 && G1Matrix[whichPeg[i]][whichPeg[j]] > 0);
-            if(!exposed && whichPeg[i] < n1 && whichPeg[j] < n1 && G2Matrix[i][j]> 0){
-                exposed = true;
+        
+        if (MultiS3::denominator_type == MultiS3::ee_global or MultiS3::denominator_type == MultiS3::rt_global){
```

Comment:
switch&case rather than long if-elseif-...

switch & case is faster than if-elseif...